Develop

pablor83 opened the pull request:
Added RequestMapping

pablor83 committed:
Created client URL

pablor83 committed:
Added RequestMapping

Elm0xz commented on src/main/java/time/TimeInCapitals/showtime/ClientShowTime.java:
```diff
@@ -0,0 +1,37 @@
+import java.net.URL;
+import java.net.URLConnection;
+
+public class ClientShowTime {
```
Ta klasa nim zdaniem wylatuje, bo to jakieś ćwiczenie którego nie potrzebujemy w końcowym kodzie

pablor83 replied:
OK wywalę, napisałem kod za pomocą którego testowałem połączenie, ale można to przecież zrobić za pomocą konsoli używając curl.

pablor83 replied:
Klasa usunięta.

Elm0xz commented on src/main/java/time/TimeInCapitals/showtime/ShowTime.java:
```diff
@@ -17,11 +20,32 @@ public String getTimeDate() {
+		return "Local Time: " + localTime + "\nLocal date: " + localDate;
+	}
+
+	@RequestMapping("/gettime")
```
Nie tak się robi metodę GET w kontrolerze

Elm0xz commented on src/main/java/time/TimeInCapitals/showtime/ShowTime.java:
```diff
@@ -17,11 +20,32 @@ public String getTimeDate() {
+
 		String localTime = localDateTime.format(timeFormat);
 		String localDate = localDateTime.format(dateFormat);
+		String locadTimeAndDate = "Local Time: " + localTime + "\nLocal date: " + localDate;
```
Literówka w zmiennej

pablor83 replied:
Poprawione

Elm0xz replied:
Nie widzę, wypchnąłeś commita?

Elm0xz commented on src/main/java/time/TimeInCapitals/showtime/ShowTime.java:
```diff
@@ -17,11 +20,32 @@ public String getTimeDate() {
 		String localDate = localDateTime.format(dateFormat);
+		String locadTimeAndDate = "Local Time: " + localTime + "\nLocal date: " + localDate;
+		
+		HashMap<String, String> timeAndDate = new HashMap<>();
```
Do czego jest tutaj potrzebna ta hashmapa? Nie wystarczy zrobić if na parametrze?

pablor83 replied:
Zastanawiałem się nad if, ale pomyślałem, że może tak będzie mnie kodu, a co za tym idzie czytelniej. Poprawię.

pablor83 replied:
Poprawiłem.

Elm0xz commented on src/main/java/time/TimeInCapitals/showtime/ShowTime.java:
```diff
@@ -17,11 +20,32 @@ public String getTimeDate() {
+		timeAndDate.put("time", localTime);
+		timeAndDate.put("date", localDate);
+				
+		return td == null ? locadTimeAndDate : timeAndDate.get(td) + " " + td;
```
Dziwna formuła zwracająca

pablor83 committed:
little corrections

-removed ClientShowTime.java
-fixed typo
-changes to the getQueryParameter method

pablor83 committed:
changes in ShowTime.java

-removed getTimeDate method
-/gettime maping name changed to /localtime

Elm0xz commented on src/main/java/time/TimeInCapitals/showtime/ShowTime.java:
```diff
@@ -13,24 +13,10 @@
-
-	@GetMapping({ "/gettime/{td}", "/gettime" })
+	
+	@GetMapping({"/", "/localtime", "/localtime/{timeOrDate}" })
```
`GetMapping("/localtime/{timeOrDate}")`

pablor83 replied:
Czyli dla każdego mapowania zrobić osobną metodę?
Przykładowo /localtime bez parametru, albo samo "/"?

Chciałbym też coś wyświetlić jeśli zostanie wpisane samo /localtime, albo "/"

Elm0xz replied:
Ok, ale po co? Lepiej zdecydować się na jednoznaczny endpoint, nigdy nie widziałem aby ktoś robił jeden endpoint mający 3 różne adresy

Ewentualnie można zrobić przekierowanie, ale to jest inna bajka

Elm0xz commented on src/main/java/time/TimeInCapitals/showtime/ShowTime.java:
```diff
@@ -13,24 +13,10 @@
+	@GetMapping({"/", "/localtime", "/localtime/{timeOrDate}" })
 	@ResponseBody
-	public String getQueryParameter(@PathVariable(required = false) String td) {
+	public String getQueryParameter(@PathVariable(required = false) String timeOrDate) throws Exception {
```
Nazwa metody wystarczy `get`

Elm0xz commented on src/main/java/time/TimeInCapitals/showtime/ShowTime.java:
```diff
@@ -13,24 +13,10 @@
 
 @Controller
 public class ShowTime {
```
Nazwę klasy zmieniamy na `LocalTimeEnpoint`

Elm0xz commented on src/main/java/time/TimeInCapitals/showtime/ShowTime.java:
```diff
@@ -13,24 +13,10 @@
-	public String getQueryParameter(@PathVariable(required = false) String td) {
+	public String getQueryParameter(@PathVariable(required = false) String timeOrDate) throws Exception {
 
 		DateTimeFormatter timeFormat = DateTimeFormatter.ofPattern("HH:mm:ss");
```
To wszystko wyłączamy do nowej klasy `LocalTimeService`

Elm0xz commented on src/main/java/time/TimeInCapitals/showtime/ShowTime.java:
```diff
@@ -43,20 +29,13 @@ public String getQueryParameter(@PathVariable(required = false) String td) {
-
-		return result;
+		if(timeOrDate == null)
+			return localTimeAndDate;
```
W każdym returnie chciałbym mieć odwołanie do oddzielnej metody z `LocalTimeService`.

Elm0xz commented:
Jeśli chodzi o architekturę - ogólnie konwencja jest taka, że w warstwie endpointów/kontrolerów zawieramy jedynie logikę "przekierowującą" flow do poszczególnych serwisów + ewentualnie mapowanie danych wejściowych. W serwisach zaś tworzymy właściwą logikę ("biznesową") programu.

pablor83 committed:
The class name has been changed and a new class has been added

pablor83 committed:
separation into three methods

pablor83 committed:
little corrections

Elm0xz commented on src/main/java/time/TimeInCapitals/showtime/LocalTimeService.java:
```diff
@@ -10,33 +10,35 @@
-	
-	private LocalDateTime localDateTime = LocalDateTime.now();
-	
+
```
Jeszcze tylko powycinaj puste linie i przepchnę pull requesta

pablor83 replied:
Zerknij czy teraz wszystko jest OK.

Elm0xz replied:
Jeśli nie robisz tego o co Cię proszę to nie jest OK. Wytnij puste linie wszędzie

pablor83 committed:
cosmetic changes

pablor83 committed:
cosmetic update
